### PR TITLE
 Merge feature/searchable-snapshots branch into 7.x (#54803) 

### PR DESCRIPTION
**Pull request by tlrx**
This is a backport of #54803 for `7.x`.

This pull request cherry picks the squashed commit from #54803 with the additional commits:
- 6f50c92 which adjusts master code to 7.x
- a114549 to mute a failing ILM test (#54818)
- 48cbca1 and 50186b2 that cleans up and unmutes the previous test
- aae12bb that adds a missing feature flag (#54861)
- 6f330e3 that adds missing serialization bits (#54864)
- bf72c02 that adjust the version in YAML tests
- a51955f that adds some plumbing for the transport client used in integration tests

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DeleteAction.java [hunk 1]
```diff
@@ -5,38 +5,63 @@
...
     public DeleteAction() {
+        this(true);
+    }
+
+    public DeleteAction(boolean deleteSearchableSnapshot) {
+        this.deleteSearchableSnapshot = deleteSearchableSnapshot;
     }
 
     public DeleteAction(StreamInput in) throws IOException {
+        if (in.getVersion().onOrAfter(Version.V_7_8_0)) {
```
Comment (tlrx) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DeleteAction.java:53]: @andreidan Version has been adjusted here

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DeleteAction.java [hunk 2]
```diff
@@ -5,38 +5,63 @@
...
+            this.deleteSearchableSnapshot = in.readBoolean();
+        } else {
+            this.deleteSearchableSnapshot = true;
+        }
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
+        if (out.getVersion().onOrAfter(Version.V_7_8_0)) {
```
Comment (tlrx) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DeleteAction.java:62]: @andreidan Version has been adjusted here

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java [hunk 3]
```diff
@@ -237,15 +237,11 @@ public void onIndexModule(IndexModule indexModule) {
 
     @Override
     public List<ActionHandler<? extends ActionRequest, ? extends ActionResponse>> getActions() {
-        if (SEARCHABLE_SNAPSHOTS_FEATURE_ENABLED) {
```
Comment (DaveCTurner) [inline x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java]: This change means that we register these actions whether the feature flag is enabled or not, which I think we should avoid. Can we revert this?

**Review by DaveCTurner — COMMENTED** [review]
I left a question about the loss of a feature flag conditional.

I'm also concerned about the number of deviations away from `List.of` -- I think this will make backporting future work much harder. Could you try and bring these more in line between the branches, ideally with `import org.elasticsearch.common.collect.List` rather than using the fully-qualified name inline?

File: x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java (hunk 3 above)
Comment (tlrx) [reply x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java]: I've been confused by the TransportClient for which I thought we should not force a system prop to be present to load the appropriate actions.

But the feature flag will be removed at some point, so we don't really care. I reverted this.

**Comment by tlrx** [comment]
@DaveCTurner Thanks for your feedback. I tried to minimize the deviations away from `List.of` event though is most cases it ends up being a complete qualification of the `org.elasticsearch.common.collect.List` class.

**Comment by DaveCTurner** [comment]
> it ends up being a complete qualification of the org.elasticsearch.common.collect.List class

Ugh, right, I see, if we also import `java.util.List` then we can't have both. Thanks anyway, I think it helps to align whitespace if nothing else.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DeleteAction.java (hunk 2 above)
Comment (andreidan) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DeleteAction.java:62]: Thanks @tlrx

**Comment by DaveCTurner** [comment]
LGTM, thanks @tlrx and @andreidan

**Review by DaveCTurner — APPROVED** [review]

**Review by andreidan — APPROVED** [review]
LGTM, great feature @tlrx @DaveCTurner

**Comment by tlrx** [comment]
Thanks David and Andrei!

**Pull request merged**